### DISCUSSION TITLE
Option to change the output directory for the ethers v5 factories

### DISCUSSION THREAD

**Discussion by PaulRBerg** [ethereum-ts/TypeChain#541] [Ideas]
I love the automatic generation of factories that is included with `@typechain/ethers-v5`. But the fact that the factories are nested in the `types` directory makes for deeply nested paths. For example, this is what I ended up writing in the usage guide for [one](https://github.com/hifi-finance/hifi/tree/3308a45a03838da9a31458ceed0aeef3a7152615/packages/amm) of the Hifi packages:

```typescript
import { HifiPool__factory } from "@hifi/amm/dist/types/factories/HifiPool__factory";
```

That is rather verbose, and in a way it feels like a contract factory should not be placed under a directory called "types". It's not a basic type - it's a class that can create objects (instances).

My feature request is to let us provide a custom path for the `factories` output directory, just like we can currently do that for the basic types (via the `outDir` config option in Hardhat).

**Comment by krzkaczor** [comment]
Factories are also exported as part of the `index.ts`, so you can import them as:

```ts
import { HifiPool__factory } from "@hifi/amm/dist/types";
```

Does this make this issue any better?

**Reply by PaulRBerg** [reply]
Kind of. But I would still prefer to import it from `@hifi/amm/dist/factories` :D

I know this is probably not easy to do, and you have other more urgent issues on your to-do list. I am just sharing my ideas re TypeChain as they come to my mind :)
